File: src/main/java/no/novari/flyt/archive/gateway/kafka/KafkaTopicProperties.java
package no.novari.flyt.archive.gateway.kafka;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

import java.time.Duration;

@Getter
@Setter
@Component
@ConfigurationProperties(prefix = "novari.flyt.archive.gateway.kafka.topic")
public class KafkaTopicProperties {

    private Duration instanceProcessingEventsRetentionTime;
}

File: src/main/kotlin/no/novari/flyt/archive/gateway/kafka/KafkaTopicProperties.kt
package no.novari.flyt.archive.gateway.kafka

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component
import java.time.Duration

@Component
@ConfigurationProperties(prefix = "novari.flyt.archive.gateway.kafka.topic")
class KafkaTopicProperties {
    var instanceProcessingEventsRetentionTime: Duration? = null
}
